packages/st2mistral/debian/mistral-server.init

```diff
@@ -67,6 +68,19 @@ do_start()
   #   1 if daemon was already running
   #   2 if daemon could not be started
   lsb_running && return 1
+
+  # /run is a tmpfs mount, so we need to check the directory exists and create.
+  # Otherwise the service will not start following a reboot.
+  if [ ! -d "$RUN_DIR" ] ; then
+      if ! mkdir -p "$RUN_DIR" ; then
+          log_failure_msg "can't create $RUN_DIR"
+          exit 1
+      fi
+      if ! chown mistral:mistral "$RUN_DIR" ; then
+          log_warning_msg "can't chown $RUN_DIR"
+      fi
+  fi
```

Comment:
This is good :+1: 
I think, we can make all that logic really simple and avoid so much verbosity.

As example, here is how `mysql` handles it: 
```sh
[ -d /var/run/mysqld ] || install -m 755 -o mysql -g root -d /var/run/mysqld
```

^^ Just a random idea.

Not sure though if `install` binary is available for all our needed platforms, but it's easy to figure out by pushing the change and check the PR status/log (our CI checks under 5 distros).